backend/src/main/java/com/auction/auctionapp/rest/controller/ProductController.java

```diff
@@ -38,6 +38,31 @@
         return ResponseEntity.ok(new ArrayList<ProductResponse>(productService.getLastChance()));
     }
 
+    @GetMapping("/products/default-sorting")
+    public ResponseEntity<List<ProductResponse>> getDefaultSortingByCategory(@RequestParam(name = "productCategoryId") int id) {
+        return ResponseEntity.ok(new ArrayList<ProductResponse>(productService.getDefaultSortingByCategory(id)));
+    }
+
+    @GetMapping("/products/new-category")
+    public ResponseEntity<List<ProductResponse>> getNewArrivalsByCategory(@RequestParam(name = "productCategoryId") int id) {
+        return ResponseEntity.ok(new ArrayList<ProductResponse>(productService.getNewArrivalsByCategory(id)));
+    }
+
+    @GetMapping("/products/last-category")
+    public ResponseEntity<List<ProductResponse>> getLastChanceByCategory(@RequestParam(name = "productCategoryId") int id) {
+        return ResponseEntity.ok(new ArrayList<ProductResponse>(productService.getLastChanceByCategory(id)));
+    }
+
+    @GetMapping("/products/high-low")
+    public ResponseEntity<List<ProductResponse>> getHighToLowByCategory(@RequestParam(name = "productCategoryId") int id) {
+        return ResponseEntity.ok(new ArrayList<ProductResponse>(productService.getHighToLowByCategory(id)));
+    }
+
+    @GetMapping("/products/low-high")
+    public ResponseEntity<List<ProductResponse>> getLowToHighByCategory(@RequestParam(name = "productCategoryId") int id) {
+        return ResponseEntity.ok(new ArrayList<ProductResponse>(productService.getLowToHighByCategory(id)));
+    }
+
```

Comment:
Instead of having multiple endpoints for every type of sort, create only one endpoint which will take in a request body based on which you will create a query to execute.

You can have some kind of query builder for this, which will take in number of parameters (from the request) and return string query